zuul-core/src/main/java/com/netflix/zuul/filters/endpoint/ProxyEndpoint.java

```diff
@@ -803,15 +803,9 @@ protected void handleOriginNonSuccessResponse(final HttpResponse originResponse,
         ClientException.ErrorType niwsErrorType;
 
         if (respStatus == 503) {
-            //Treat 503 status from Origin similarly to connection failures, ie. we want to back off from this server
             statusCategory = FAILURE_ORIGIN_THROTTLED;
             niwsErrorType = ClientException.ErrorType.SERVER_THROTTLED;
             obe = new OutboundException(OutboundErrorType.SERVICE_UNAVAILABLE, requestAttempts);
-            if (originConn != null) {
-                originConn.getServer().incrementSuccessiveConnectionFailureCount();
-                originConn.getServer().addToFailureCount();
```

Comment:
Think we should still track the failure counter here? Otherwise we are missing the server stats on failures

Reply:
I think these particular failure counters are intended to track server errors, e.g., connectivity problems. If we don't want 503 to fall into that category anymore, we probably don't want to avoid these servers in the future.

Reply:
🚀